tests/expected/cpp/34327-Issue_3044.cpp

```diff
@@ -0,0 +1,61 @@
+int main()
+{
+	int af;
+	int A;
+	int B;
+	switch (af)
+	{
+		case 1:
+		{
+			B = 2;
+		}
+
+		case 2:
+		{
+			return 1;
+		}
+
+		case 3:
+		{
+			A = 1;
+			break;
+#ifdef ALL_THE_CASE
+		}
+
+		case 4:
+		{
+			return 2;
+#endif
+#ifdef ALL_THE_CASE
+		}
+
+		case 5:
+		{
+			B = 2;
+#endif
```

Comment:
Althought technically speaking this results in correct code, it definitely looks a bit weird in the way the case is broken up.